VotingApplication/VotingApplication.Web/Api/Controllers/DashboardController.cs

```diff
@@ -0,0 +1,151 @@
+    {
+        public DashboardController()
+        {
+        }
+
+        public DashboardController(IContextFactory contextFactory)
+            : base(contextFactory)
+        {
+        }
+
+        [HttpGet]
+        [Authorize]
+        public List<DashboardPollResponseModel> Polls()
+        {
+            using (IVotingContext context = _contextFactory.CreateContext())
+            {
+                string userId = User.Identity.GetUserId();
+
+                IEnumerable<DashboardPollResponseModel> userPolls = context
+                    .Polls
+                    .Where(p => p.CreatorIdentity == userId)
+                    .OrderByDescending(p => p.CreatedDate)
+                    .Select(CreateDashboardResponseFromModel);
+
+                return userPolls.ToList();
+            }
+        }
+
+        private static DashboardPollResponseModel CreateDashboardResponseFromModel(Poll poll)
+        {
+            return new DashboardPollResponseModel()
+            {
+                UUID = poll.UUID,
+                ManageId = poll.ManageId,
+                Name = poll.Name,
+                Creator = poll.Creator,
+                CreatedDate = poll.CreatedDate,
+                ExpiryDate = poll.ExpiryDate
+            };
+        }
+
+
+        [HttpPost]
+        [Authorize]
+        public CopyPollResponseModel Copy(CopyPollRequestModel pollCopyRequest)
+        {
+            if (pollCopyRequest == null)
+            {
+                this.ThrowError(HttpStatusCode.BadRequest);
+            }
+
+            if (!ModelState.IsValid)
+            {
+                this.ThrowError(HttpStatusCode.BadRequest, ModelState);
+            }
+
+            using (IVotingContext context = _contextFactory.CreateContext())
+            {
+                string userId = User.Identity.GetUserId();
+
+                Poll pollToCopy = context
+                    .Polls
+                    .Include(p => p.Options)
+                    .SingleOrDefault(p => p.UUID == pollCopyRequest.UUIDToCopy);
+
+                if (pollToCopy == null)
+                {
+                    this.ThrowError(HttpStatusCode.BadRequest);
+                }
+
+
+                if (pollToCopy.CreatorIdentity != userId)
+                {
+                    this.ThrowError(HttpStatusCode.Forbidden);
+                }
+
+                Poll newPoll = CopyPoll(pollToCopy);
+
+                context.Polls.Add(newPoll);
+                context.SaveChanges();
+
+                return new CopyPollResponseModel()
+                {
+                    newPollId = newPoll.UUID,
+                    newManageId = newPoll.ManageId
+                };
+            }
+        }
+
+        private static Poll CopyPoll(Poll pollToCopy)
+        {
+            return new Poll()
+            {
+                UUID = Guid.NewGuid(),
+                ManageId = Guid.NewGuid(),
+                Name = "Copy of " + pollToCopy.Name,
```

Comment:
Not sure we should have the "Copy of", as the poll name isn't unique. If i'm copying "Friday Lunch", I would expect another "Friday Lunch"